src/storage/src/hummock/compactor/compactor_runner.rs

```diff
@@ -716,7 +716,7 @@ where
             last_key.is_empty() || iter_key.user_key != last_key.user_key.as_ref();
 
         let mut drop = false;
-        let epoch = iter_key.epoch;
+        let epoch = iter_key.epoch_with_gap.get_epoch();
```

Comment:
Here we may fetch the pure epoch instead of the epoch with gap to be consistent with the original logic.